targets/targets.json

```diff
@@ -7133,7 +7138,8 @@
         "macros_remove": ["MBEDTLS_CONFIG_HW_SUPPORT"],
         "overrides": {
             "lf_clock_src": "NRF_LF_SRC_RC"
-        }
+        },
+		"device_has_remove": ["RESET_REASON"]
```

Comment:
Why not just remove `RESET_REASON` from `MCU_NRF52840` ?

Reply:
NRF52840_DK this device also inherits from MCU_NRF52840 but this one have support for reset reason